diff --git a/org.adempiere.ui.swing/META-INF/MANIFEST.MF b/org.adempiere.ui.swing/META-INF/MANIFEST.MF
--- a/org.adempiere.ui.swing/META-INF/MANIFEST.MF
+++ b/org.adempiere.ui.swing/META-INF/MANIFEST.MF
@@ -78,7 +78,8 @@
 Eclipse-ExtensibleAPI: true
 Eclipse-RegisterBuddy: org.adempiere.base
 Bundle-RequiredExecutionEnvironment: JavaSE-1.6
-Import-Package: javax.net.ssl,
+Import-Package: cz.muni.fi.vavmar.printeditor,
+ javax.net.ssl,
  org.apache.ecs,
  org.apache.ecs.xhtml,
  org.apache.poi.hssf.usermodel,
diff --git a/org.adempiere.ui.swing/src/org/compiere/apps/APanel.java b/org.adempiere.ui.swing/src/org/compiere/apps/APanel.java
--- a/org.adempiere.ui.swing/src/org/compiere/apps/APanel.java
+++ b/org.adempiere.ui.swing/src/org/compiere/apps/APanel.java
@@ -54,6 +54,7 @@
 import javax.swing.Action;
 import javax.swing.BorderFactory;
 import javax.swing.ImageIcon;
+import javax.swing.JButton;
 import javax.swing.JDialog;
 import javax.swing.JFrame;
 import javax.swing.JList;
@@ -124,6 +125,8 @@
 import org.compiere.util.NamePair;
 import org.compiere.util.Util;
 
+//VavMar Edited
+import cz.muni.fi.vavmar.printeditor.MainFrame;
 /**
  *	Main Panel of application window.
  *  <pre>
@@ -324,7 +327,7 @@
 							aExport = null,
 							aRefresh, aHistory, aAttachment, aChat, aMulti, aFind,
 							aWorkflow, aZoomAcross, aRequest, aWinSize, aArchive,
-							aProcess;
+							aProcess, printEditorButton;
 	/** Ignore Button		*/
 	public AppsAction		aIgnore;
 	/** Save Button			*/
@@ -361,6 +364,10 @@
 		aReport = 	addAction("Report",			mFile, 	KeyStroke.getKeyStroke(KeyEvent.VK_F11, 0),	false);
 		aPrint = 	addAction("Print",			mFile, 	KeyStroke.getKeyStroke(KeyEvent.VK_F12, 0),	false);
 		aPrintPreview = addAction("PrintPreview",	mFile, KeyStroke.getKeyStroke(KeyEvent.VK_P, Event.SHIFT_MASK+Event.ALT_MASK), false);
+		
+		//Edited VavMAr
+		printEditorButton = addAction("PrintEditor", mFile, KeyStroke.getKeyStroke(KeyEvent.VK_P, Event.CTRL_MASK + Event.ALT_MASK), false);
+		
 		if (MRole.getDefault().isCanExport())
 		{
 			aExport = addAction("Export", mFile, null, false);
@@ -540,6 +547,9 @@
 		toolBar.add(aArchive.getButton());
 		toolBar.add(aPrintPreview.getButton());
 		toolBar.add(aPrint.getButton());
+		toolBar.addSeparator();
+		toolBar.add(printEditorButton);
+		
 		// FR [ 1757088 ]
 		if((m_curGC == null) || (m_curGC != null && !m_curGC.isDetailGrid())){
 			toolBar.addSeparator();
@@ -1731,6 +1741,12 @@
 				cmd_print();
 			else if (cmd.equals(aPrintPreview.getName()))
 				cmd_print(true);
+			else if (cmd.equals(printEditorButton.getName())){
+				//VavMar Edited
+				log.log(Level.WARNING, "Kliknuto na tlacitko pro spusteni PrintEditoru.");
+				JOptionPane.showConfirmDialog(null, "Kliknuto na tlacitko pro spusteni PrintEditoru.");
+				MainFrame.runProgram();
+			}
 			else if (aExport != null && cmd.equals(aExport.getName()))
 				cmd_export();
 			else if (cmd.equals(aEnd.getName()))
